src/shared_symbol_table.rs

```diff
@@ -0,0 +1,54 @@
+use crate::result::illegal_operation;
+use crate::IonResult;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+/// Stores [SharedSymbolTable] with the table name, version and imports
+/// For more information on [SharedSymbolTable]: https://amzn.github.io/ion-docs/docs/symbols.html#shared-symbol-tables
+pub struct SharedSymbolTable {
+    name: String,
+    version: usize,
+    imports: Vec<Option<String>>,
+}
+
+impl SharedSymbolTable {
+    pub fn new(
+        name: String,
+        version: Option<usize>,
```

Comment:
I think this can just be `version: usize`. Finding an `import` struct that doesn't specify a `version` field happens at a lower layer than creating a `SharedSymbolTable` instance in memory; the reader can handle defaulting the `version` to 1 if it's missing.

Reply:
Understood. I will change this with next commit.